src/Data/Monsters/Monster.cs

```diff
@@ -26,6 +43,210 @@ public Monster(MonsterType type)
+        /// Updates monster AI behavior based on current state and nearby characters
+        /// </summary>
+        public void Update()
+        {
+            if (IsDead) return;
+            if (_currentMap == null) return;
+
+            switch (State)
+            {
+                case MonsterState.Sleeping:
+                    CheckForAggroTrigger();
+                    break;
+                case MonsterState.Awake:
+                case MonsterState.Chasing:
+                    UpdateChasing();
+                    break;
+                case MonsterState.Attacking:
+                    UpdateAttacking();
+                    break;
+            }
+        }
+
+        /// <summary>
+        /// Checks if any character is within aggro radius and wakes up if so
+        /// </summary>
+        private void CheckForAggroTrigger()
+        {
+            if (_currentMap == null) return;
+            
+            var characters = _currentMap.Characters.Where(c => !c.IsDead);
+            foreach (var character in characters)
+            {
+                float distance = Vector2.Distance(Position, character.Position);
+                if (distance <= AggroRadius)
+                {
+                    WakeUp(character);
+                    break;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Wakes up this monster and nearby monsters, starts chasing the target
+        /// </summary>
+        public void WakeUp(Character trigger)
+        {
+            if (State == MonsterState.Sleeping)
+            {
+                State = MonsterState.Awake;
+                Target = trigger;
+                
+                // Wake up nearby monsters (chain awakening)
+                WakeUpNearbyMonsters();
+                
+                // Immediately start chasing
+                State = MonsterState.Chasing;
+            }
+        }
+
+        /// <summary>
+        /// Wakes up other sleeping monsters within a certain radius
+        /// </summary>
+        private void WakeUpNearbyMonsters()
+        {
+            if (_currentMap == null) return;
+            
+            float wakeUpRadius = 150f; // Slightly larger than aggro radius
+            var nearbyMonsters = _currentMap.Monsters.Where(m => m != this && m.State == MonsterState.Sleeping);
+            
+            foreach (var monster in nearbyMonsters)
+            {
+                float distance = Vector2.Distance(Position, monster.Position);
+                if (distance <= wakeUpRadius)
+                {
+                    monster.State = MonsterState.Awake;
+                    monster.Target = Target; // Share the same target
+                    monster.State = MonsterState.Chasing;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Updates chasing behavior - moves toward target and attacks when in range
+        /// </summary>
+        private void UpdateChasing()
```

Comment:
I don't think it is necessary to have a chasing state that is distinct from the attack state as both are really similar (move until in range of the ability, then attack).